core/src/main/java/io/confluent/kafka/schemaregistry/storage/KafkaStoreReaderThread.java

```diff
@@ -147,9 +141,9 @@ public void doWork() {
           V message = null;
           try {
             message =
-                messageBytes == null ? null : serializer.deserializeValue(messageKey, messageBytes);
+                record.value() == null ? null : serializer.deserializeValue(messageKey, record.value());
           } catch (SerializationException e) {
-            log.error("Failed to deserialize a schema or config update", e);
+            throw new RuntimeException("Failed to deserialize a schema or config update", e);
```

Comment:
Sorry, what I meant is log and continue to the next message. The problem previously was that we caught the error and then proceeded with the logic below, which seems like it would incorrectly result in a deletion.